contrib/ansible/vagrant/Vagrantfile

```diff
@@ -1,46 +1,100 @@
 # -*- mode: ruby -*-
 # vi: set ft=ruby :
+require 'date'
 
-# All Vagrant configuration is done below. The "2" in Vagrant.configure
-# configures the configuration version (we support older styles for
-# backwards compatibility). Please don't change it unless you know what
-# you're doing.
-Vagrant.configure(2) do |config|
-  config.vm.box = "chef/centos-7.0"
+### This is a new provider, different then cloudbau's.  
+### RUN: vagrant plugin uninstall vagrant-openstack-plugin"
+### Then RUN: "vagrant plugin install vagrant-openstack-provider"
+require 'vagrant-openstack-provider'
 
-#  config.vm.network "public_network"
+Total=3
 
-  config.vm.define "master", primary: true do |master|
-    master.vm.hostname = "master.vms.local"
-    master.vm.network "private_network", ip: "192.168.1.100"
+VAGRANTFILE_API_VERSION = "2"
+
+# Openstack + Hostmanager providers are best used with latest versions.
+Vagrant.require_version ">= 1.7"
+
+### If you want to change variables in all.yml, use this snippet.
+### Just add a new line below as necessary...
+text = File.read('../group_vars/all.yml')
+new_contents = text.gsub("#ansible_ssh_user: root", "ansible_ssh_user: fedora")
```

Comment:
So I guess back to original concern with this Vagrantfile. I like having multiple options for providers but now with this as the only and default provider users can not easily create a cluster.  I would like to have this Vagrant file in a directory so we can orginize the providers. `contrib/ansible/vagrant/openstack/Vagrantfile` I have a WIP Vagrantfile with atomic as well that i would like to get working one of these days. I do not like the fact that i now have to have openstack running to get a simple cluster going, let alone an account.


Now if we want this to be the actual deployment mechanism then that would change a lot of things. That is what we are building the Ansible for. I would prefer not to have Vagrant be the deployment base in that case as well. Similar concept to how bash is for the saltbase today in Kubernetes.